### ISSUE TITLE
Create a mechanism for adding notes to the changelog

### ISSUE THREAD

**Issue by zephraph** [intuit/auto#375]
**Is your feature request related to a problem? Please describe.**

Currently the changelog is generated by PR titles. I think for the vast majority of use cases this is sufficient. In some cases though, there may be a need to add context to the release.

An example would be for breaking changes. In those circumstances, having some sort of migration guide as a part of the changelog/release would be beneficial. 

**Describe the solution you'd like**

I'm not 100% on the setup here, but it could be as simple as having a release notes section in your PR description

```
# Release Notes

Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor 
incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud 
exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat.
```

or maybe a `details` tag

```
<details>
  <summary>Release Notes</summary

Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor 
incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud 
exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat.

<details>
```

In the output it could have a top level heading like `### Release Notes` and it might be useful to separate each set of docs by PR. 

The final rendered output of the changelog might look something like the below: 

---

# v5.0.1 (Sat May 04 2019)

### Release Notes

_From #371_

`auto shipit` will can now only be ran on the base branch ([which is configurable]()). If you want to ship a version in the PR, you can now look to using [canary releases]().

---

#### 🐛  Bug Fix

- shipit will publish a canary locally when not on master [#371](https://github.com/intuit/auto/pull/371) ([@hipstersmoothie](https://github.com/hipstersmoothie))

#### Authors: 1

- Andrew Lisowski ([@hipstersmoothie](https://github.com/hipstersmoothie))

**Comment by hipstersmoothie** [comment]
> An example would be for breaking changes. In those circumstances, having some sort of migration guide as a part of the changelog/release would be beneficial. 

I've been trying to include things like this in the actual PR for now. I really like the solution you describe though!

I like the section better than the details tag.

**Comment by zephraph** [comment]
This would do both 😄 . For things like slack integration or renovate updates having it in the changelog would make the changes more visible. 

The section is definitely less cumbersome to write, but you'd still need to figure out what goes in the section. That feels a little harder.

**Comment by hipstersmoothie** [comment]
Another option

```md
> # Release notes
> Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor 
> incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud 
> exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat.
```

> # Release notes
> Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor 
> incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud 
> exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat.

**Comment by zephraph** [comment]
The only thing about that is it's harder to type. You have to add a `>` for every line, which is someone cumbersome. 

For the section matching, the algo could be something like...

1. Regex for `/(#{1,5}) Release Notes/` (ish)
2. Use the `#` capture as the break condition
3. If you encounter a line that starts with that many `#` _or fewer_ finish the notes capture

It's not perfect, but it'd be easier to write on the user's side. We can always iterate on it in the future.

**Comment by hipstersmoothie** [comment]
@zephraph I've implemented the algo you detailed. Wasn't too many changes. I think this meets the use case well now.

https://github.com/intuit/auto/pull/380/files#diff-fb34033d1fd5720d73c88aaef9f8d9a0

**Comment by zephraph** [comment]
You're a wizard

**Comment by adierkens** [comment]
:rocket: Issue was released in v6.3.0 :rocket:
